### DISCUSSION TITLE
Where are the preferences stored?

### DISCUSSION THREAD

**Discussion by chrisgrieser** [rxhanson/Charmstone-Community#9] [Q&A]
First of all, thank you for this wonderful app. I've been looking for something like this for a while now, and only stumbled open this by chance.

I plan to use Charmstone on multiple devices and would like to sync, or at least copy over configuration, to save some time. However, as opposed to most other apps, I couldn't find a `.plist` file belonging to Charmstone in `~/Library/Preferences/`? Does it have an unusual name or am I simply missing something here?

**Comment by rxhanson** [comment]
Charmstone is sandboxed :)

`~/Library/Containers/com.knollsoft.Charmstone/Data/Library/Preferences/com.knollsoft.Charmstone.plist`

(I had to look it up: https://stackoverflow.com/questions/7982740/where-is-a-mac-applications-nsuserdefaults-data-stored)

**Reply by chrisgrieser** [reply]
hmmm, I think there was [some issue with preferences from sandboxed apps being not easily copied](https://apple.stackexchange.com/questions/383306/how-to-create-a-custom-readable-symlink-out-of-a-sandboxed-container), but I'll give it a try when I am on the other device again.

Nevertheless, iCloud syncing would be fantastic to save one from that hassle. But I can understand that cloud data syncing for such a relatively small amount of data is a bit much effort.

Maybe an option to export the settings, so you can import them at a different device? 🤔

**Reply by rxhanson** [reply]
Good to know, I was unaware of that limitation here.

iCloud sync & export/import are on my backlog, but I don't have a solidified timeframe for when I would be adding either of those in.

Personally, I use [prefs editor](https://apps.tempel.org/PrefsEditor/) for all of my defaults related stuff like this. Obviously, that's not an excellent solution for this use case, but it could make things a little easier until I get something out the door.
